src/bin/cargo/commands/owner.rs

```diff
@@ -1,51 +1,149 @@
+                        Arg::new("crate")
+                            .value_name("CRATE_NAME")
+                            .help("Crate name that you want to manage the owner"),
+                    ]),
+            ),
+            add_arg(
+                Command::new("remove")
+                    .about("Name of a user or team to remove as an owner")
+                    .arg_quiet()
+                    .args([
+                        Arg::new("removeowner")
+                            .required(true)
+                            .value_delimiter(',')
+                            .value_name("OWNER_NAME")
+                            .help("Name of the owner you want to remove"),
+                        Arg::new("crate")
+                            .value_name("CRATE_NAME")
+                            .help("Crate name that you want to manage the owner"),
+                    ]),
+            ),
+            add_arg(
+                Command::new("list")
+                    .about("List owners of a crate")
+                    .arg_quiet()
+                    .arg(
+                        Arg::new("crate")
+                            .value_name("CRATE_NAME")
+                            .help("Crate name which you want to list all owner names"),
+                    ),
+            ),
+        ])
         .arg(opt("index", "Registry index to modify owners for").value_name("INDEX"))
         .arg(opt("token", "API token to use when authenticating").value_name("TOKEN"))
         .arg(opt("registry", "Registry to use").value_name("REGISTRY"))
         .after_help("Run `cargo help owner` for more detailed information.\n")
 }
 
+fn add_arg(com: Command) -> Command {
+    com.arg(opt("index", "Registry index to modify owners for").value_name("INDEX"))
+        .arg(opt("token", "API token to use when authenticating").value_name("TOKEN"))
+        .arg(opt("registry", "Registry to use").value_name("REGISTRY"))
+}
+
 pub fn exec(config: &mut Config, args: &ArgMatches) -> CliResult {
-    let registry = args.registry(config)?;
+    let command = args.subcommand();
+
+    let (sc, krate, token, index, registry) = if command.is_some() {
+        let (sc, arg) = command.unwrap();
+        let subcommand = match sc {
+            "list" => SubCommand::List,
+            "add" => SubCommand::Add,
+            "remove" => SubCommand::Remove,
+            _ => SubCommand::None,
+        };
+        (
+            subcommand,
+            arg.clone().get_one::<String>("crate").cloned(),
+            arg.get_one::<String>("token").cloned().map(Secret::from),
+            arg.get_one::<String>("index").cloned(),
+            arg.registry(config)?,
+        )
+    } else {
+        (
+            SubCommand::None,
+            args.clone().get_one::<String>("crate").cloned(),
+            args.get_one::<String>("token").cloned().map(Secret::from),
+            args.get_one::<String>("index").cloned(),
+            args.registry(config)?,
```

Comment:
Returning long tuples is a bit hard to read and the pattern for how this code operations isn't the most straightforward.

Instead what you can do is
```rust
let (to_add, to_remove, to_list) = match args.subcommand() {
    Some(("add", args)) => {
        ...
    },
    ...
    Some((name, _)) => {
        unreachable!("clap should prevent {name} from being used")
    },
    None = {
        ...
    }
};

let common_args = args.subcommand().map(|(_, args)| args).unwrap_or(args);
...
let opts OwnerOptions {
    krate: common_args.clone().get_one::<String>("crate").cloned(),
    ...
}
```

Or to make the code more future proof and have no tuple, just create `OwnerOptions` in reach branch of the `match`.  The duplicated code is trivial and a lot of it would be compiler enforced

Reply:
Thanks for the reminder, maybe this is the best way.